crates/hyperswitch_connectors/src/connectors/santander/transformers.rs

```diff
@@ -435,25 +867,70 @@ impl<F, T> TryFrom<ResponseRouterData<F, SantanderPaymentsSyncResponse, T, Payme
         item: ResponseRouterData<F, SantanderPaymentsSyncResponse, T, PaymentsResponseData>,
     ) -> Result<Self, Self::Error> {
         let response = item.response.clone();
-        let connector_metadata = response.pix.first().map(|pix| {
-            serde_json::json!({
-                "end_to_end_id": pix.end_to_end_id.clone().expose()
-            })
-        });
-        Ok(Self {
-            status: common_enums::AttemptStatus::from(item.response.status),
-            response: Ok(PaymentsResponseData::TransactionResponse {
-                resource_id: ResponseId::ConnectorTransactionId(response.transaction_id.clone()),
-                redirection_data: Box::new(None),
-                mandate_reference: Box::new(None),
-                connector_metadata,
-                network_txn_id: None,
-                connector_response_reference_id: None,
-                incremental_authorization_allowed: None,
-                charges: None,
-            }),
-            ..item.data
-        })
+
+        match response {
+            SantanderPaymentsSyncResponse::PixQRCode(pix_data) => {
+                let connector_metadata = pix_data.pix.first().map(|pix| {
+                    serde_json::json!({
+                        "end_to_end_id": pix.end_to_end_id.clone().expose()
+                    })
+                });
+                Ok(Self {
+                    status: common_enums::AttemptStatus::from(pix_data.status),
+                    response: Ok(PaymentsResponseData::TransactionResponse {
```

Comment:
is it anywhere mentioned in the docs - that the connector can throw 400 for failure? If so, it needs to be handled in all the responses in the transformers file, otherwise the current 
 ErrorHandling should be enough